app/models/task.rb

```diff
@@ -682,6 +683,15 @@ def set_assigned_at
     self.assigned_at = created_at unless assigned_at
   end
 
+  def set_dispatch_blocker_parent
```

Comment:
When action is conditional, I've seen `conditionally_` as a prefix.
```suggestion
  def conditionally_set_dispatch_blocker_parent
```

Alternative, is to make the `if` on line 34 be `if: [:blocking_dispatch?, :dispatch_task_exists?]`.